pkg/kube/events.go

```diff
@@ -0,0 +1,39 @@
+package kube
+
+import (
+	"github.com/golang/glog"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/event"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+func NewEventRecorder(namespace string, cli kubernetes.Interface) record.EventRecorder {
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(
+		&typedcorev1.EventSinkImpl{
+			Interface: cli.CoreV1().Events(namespace)})
+	eventRecorder := eventBroadcaster.NewRecorder(
+		scheme.Scheme,
+		v1.EventSource{
+			Component: types.ControllerServiceName})
+	return eventRecorder
+}
+
+func RecordWarningEvent(namespace string, reason string, message string, recorder record.EventRecorder, cli kubernetes.Interface) {
+	event.Recordf(reason, message)
+	deployment, _ := cli.CoreV1().Services(namespace).Get(types.ControllerServiceName, metav1.GetOptions{})
```

Comment:
Since the deployment variable is always the same, why not retrieving it in the `NewEventRecorder` instead?
One issue here, is that the variable is named `deployment` but it is actually retrieving a `Service`.

Reply:
Nice catch! I will change the name and I will try to not get the service each time that a event needs to be sent.